src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,53 @@
+package seedu.contactbook.data.person;
+
+import seedu.contactbook.data.exception.IllegalValueException;
+
+public class Contact {
```

Comment:
A few issues for you to consider:
- Is the MESSAGE_CONTACT_CONSTRAINTS and CONTACT_VALIDATION_REGEX common for all subclass? If not, how do we ensure the subclass provides them?